DXMainClient/DXGUI/Multiplayer/CnCNet/ChoiceNotificationBox.cs

```diff
@@ -0,0 +1,193 @@
+﻿using ClientCore;
+using ClientGUI;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Rampastring.XNAUI;
+using Rampastring.XNAUI.XNAControls;
+using System;
+
+namespace DTAClient.DXGUI.Multiplayer.CnCNet
+{
+    /// <summary>
+    /// A box that allows users to make a choice,
+    /// top-left of the game window.
+    /// </summary>
+    public class ChoiceNotificationBox : XNAPanel
+    {
+        const double DOWN_TIME_WAIT_SECONDS = 4.0;
+        const double DOWN_MOVEMENT_RATE = 2.0;
+        const double UP_MOVEMENT_RATE = 2.0;
+
+        public ChoiceNotificationBox(WindowManager windowManager) : base(windowManager)
+        {
+            downTimeWaitTime = TimeSpan.FromSeconds(DOWN_TIME_WAIT_SECONDS);
+        }
+
+        public Action<ChoiceNotificationBox> AffirmativeClickedAction { get; set; }
+        public Action<ChoiceNotificationBox> NegativeClickedAction { get; set; }
+
+        XNALabel lblHeader;
+        XNAPanel gameIconPanel;
+        XNALabel lblSender;
+        XNALabel lblChoiceText;
+        XNAClientButton affirmativeButton;
+        XNAClientButton negativeButton;
+
+        TimeSpan downTime = TimeSpan.Zero;
+
+        TimeSpan downTimeWaitTime;
+
+        bool isDown = false;
+
+        const int boxHeight = 101;
+
+        double locationY = -boxHeight;
+
+        public override void Initialize()
+        {
+            Name = "ChoiceNotificationBox";
```

Comment:
Use `nameof` (in this case, `Name = nameof(ChoiceNotificationBox)`). The same applies to all child controls.

Reply:
Done.